src/Parsers/ExpressionListParsers.cpp

```diff
@@ -2032,7 +2034,15 @@ std::unique_ptr<Layer> getFunctionLayer(ASTPtr identifier, bool is_table_functio
     else if (function_name_lowercase == "grouping")
         return std::make_unique<FunctionLayer>(function_name_lowercase, allow_function_parameters_);
     else
-        return std::make_unique<FunctionLayer>(function_name, allow_function_parameters_);
+    {
+        bool has_database_name = false;
+        if (const auto *ast_identifier  = identifier->as<ASTIdentifier>())
```

Comment:
Here check `if (const auto *ast_identifier = identifier->as<ASTIdentifier>())` will always be true (otherwise above `String function_name = getIdentifierName(identifier);` will throw exception), so probably `assert_cast` is better here or we can even do without it - just `identifier->as<ASTIdentifier>()` without check as it is already checked at this point.

Reply:
good point, addressed it.